src/s3/commands/presignedURL.ts

```diff
@@ -41,33 +47,115 @@ export async function presignedURLCommand(
+        this.form.signedUrlParams.bucketName.bindPrompter(({ region }) =>
+            createBucketPrompter(assertDefined(region, 'region'))
+        )
+
+        if (node) {
+            this.form.signedUrlParams.key.setDefault(node.file.key)
+        }
+
+        this.form.signedUrlParams.key.bindPrompter(
+            ({ region, signedUrlParams }) =>
+                createS3FilePrompter(
+                    assertDefined(region, 'region'),
+                    assertDefined(signedUrlParams?.bucketName, 'bucketName'),
+                    assertDefined(signedUrlParams?.operation, 'operation')
+                ),
+            node ? { showWhen: state => state.signedUrlParams?.operation === 'putObject' } : undefined
+        )
 
-    const time = Number(timeStr)
+        this.form.signedUrlParams.time.bindPrompter(() => createExpiryPrompter().transform(s => Number(s) * 60))
 
-    if (isNaN(time) || time < 0) {
-        throw new ToolkitError('The provided expiration time is not a positive number', { code: 'InvalidInput' })
+        function assertDefined<T>(val: T | undefined, key: string): T {
+            if (val === undefined) {
+                throw Error(`PresignedUrlWizard: "${key}" is undefined`)
+            }
+            return val
+        }
     }
+}
+
+function createOperationPrompter() {
+    const items: DataQuickPickItem<string>[] = [
+        { label: 'Download (GET)', data: 'getObject' },
+        { label: 'Upload (PUT)', data: 'putObject' },
+    ]
 
-    return time
+    return createQuickPick(items, { title: 'Presigned URL: Choose an operation', buttons: createCommonButtons() })
 }
 
-function validateTime(time: string): string | undefined {
-    const number = Number(time)
+function createExpiryPrompter() {
+    return createInputBox({
+        value: '15',
+        prompt: 'Specify the expiry time (minutes) for the URL',
+        placeholder: 'Defaults to 15 minutes',
+        validateInput: validateTime,
+        buttons: createCommonButtons(),
+    })
+}
 
-    if (isNaN(Number(time)) || !Number.isSafeInteger(number) || Number(time) <= 0) {
-        return invalidNumberWarning
+function createBucketPrompter(region: string) {
+    const client = new DefaultS3Client(region)
+    const items = client.listBucketsIterable().map(b => [
+        {
+            label: b.Name,
+            data: b.Name,
+        },
+    ])
+
+    return createQuickPick(items, { title: 'Select an S3 Bucket', buttons: createCommonButtons() })
+}
+
+function createS3FilePrompter(region: string, bucket: string, operation: string) {
+    if (operation === 'getObject') {
+        const items = getS3Files(region, bucket)
+        return createQuickPick(items, {
+            title: 'Choose the file for the presigned URL',
+            buttons: createCommonButtons(),
+        })
     }
+    return createInputBox({
+        title: 'Create a key',
+        buttons: createCommonButtons(),
```

Comment:
"Create a key" is a bit obscure. This might be more intuitive (also uses the same "Specify" verb that was used for the "Specify the time..." message; using a verb like "Create" implies different semantics):

    Specify the key (S3 file path) where the upload will be saved

Also let's link to https://docs.aws.amazon.com/AmazonS3/latest/userguide/PresignedUrlUploadObject.html from the help button in this step.